### ISSUE TITLE
2016.3.0 FreeBSD lost cpu_flags and cpu_model

### ISSUE THREAD

**Issue by ohauer** [saltstack/salt#33608]
### Description of Issue/Question
2016.3.0 no longer calls _bsd_cpudata for FreeBSD (only for OpenBSD and NetBSD)
I haven't found the exact commit, but can show the issue as diff.

### diff shows the issue
```
--- salt-2015.8.10/salt/grains/core.py
+++ salt-2016.3.0/salt/grains/core.py
@@ -1324,7 +1324,14 @@
         grains.update(_osx_gpudata())
     else:
         grains['os'] = grains['kernel']
-    if grains['kernel'] in ('FreeBSD', 'OpenBSD', 'NetBSD'):
+    if grains['kernel'] == 'FreeBSD':
+        try:
+            grains['osrelease'] = __salt__['cmd.run']('freebsd-version -u').split('-')[0]
+        except salt.exceptions.CommandExecutionError:
+            # freebsd-version was introduced in 10.0.
+            # derive osrelease from kernelversion prior to that
+            grains['osrelease'] = grains['kernelrelease'].split('-')[0]
+    if grains['kernel'] in ('OpenBSD', 'NetBSD'):
         grains.update(_bsd_cpudata(grains))
         grains['osrelease'] = grains['kernelrelease'].split('-')[0]
         if grains['kernel'] == 'NetBSD':
```

**Comment by ohauer** [comment]
This should bring back the missing cpu info on FreeBSD
```
--- salt-2016.3.0/salt/grains/core.py.orig
+++ salt-2016.3.0/salt/grains/core.py  
@@ -1325,6 +1325,7 @@
     else:
         grains['os'] = grains['kernel']
     if grains['kernel'] == 'FreeBSD':
+        grains.update(_bsd_cpudata(grains))
         try:
             grains['osrelease'] = __salt__['cmd.run']('freebsd-version -u').split('-')[0]
         except salt.exceptions.CommandExecutionError:
```

**Comment by Ch3LL** [comment]
@ohauer did you test the addition of `grains.update`? I am seeing the same behavior where cpu_flags an d cpu_model are not available but adding that line did not resolve the issue. Although I am testing with salt-ssh.

**Comment by ohauer** [comment]
@Ch3LL yes, and it works for me.
In case you have modified "core.py" after installation, do not forget to remove core.pyc

**Comment by amontalban** [comment]
Hey guys,

Do you know if this is going to get merged 2016.3.2?

Let me know if you need help to test this bug.

Thanks!

**Comment by junovitch** [comment]
Do we just need a pull request to accept for this?  I added @ohauer's patch back to the FreeBSD Salt port in https://github.com/freebsd/freebsd-ports/commit/02145140146049128f322fe093835cffec47334c

**Comment by junovitch** [comment]
Completely didn't notice this was already added in commit https://github.com/saltstack/salt/commit/f6dfbac5caa07e2d4aa0e2d4c05869b4f389864c

**Comment by rallytime** [comment]
@junovitch and @amontalban - Yes, the fix mentioned above is in pull request #34565, which will be in the 2016.3.2 release. Are we good to close up this issue?
